8.x/theme-development/products.md

```diff
@@ -1,3 +1,435 @@
+
+**NOTE:** `product` is a Laravel model, extends from Laravel `BaseModel`.
+
+##### SOURCE
+
+```php
+class product extends BaseModel implements
+```
+
+
+A product is related to objects, Following, you can see these relationships in the table. 
+
+Num   |Property              | Datatype                         |Description
+------|----------------------|----------------------------------|------------------------------------------
+**1** | `directory`            |Directory                         |Specifies what category this product is in.
+**2** | `directories`          |Directory[]                       |There is a list of parent directories that indicate in which other directories this product is manufactured, or (Short link of goods in other directories).
+**3** | `productStructure`     |ProductStructure                  |Displays the overall structure of a product.
+**4** | `colors`               |Color[]                           |List of product colors.
+**5** | `prices`               |ProductPrice[]                    |List of previous product prices.
+**6** | `images`               |ProductImage[]                    |List of product images.
+**7** | `attributeKeys`        |ProductStructureAttributeKey[]    |List of keys as product features.
+**8** | `attributeValues`      |ProductStructureAttributeValue[]  |Values ​​that each product has recorded for itself for each key.
+**9** | `pAttributes`          |ProductAttribute[]                |Holds `attributeKeys` and` attributeValues` assignments as a relation.
+**10**| `invoices`             |Invoice[]                         |Specifies what factors the product is used for.
+**11**| `wishLists`            |CustomerUser[]                    |The list that the user wants to create in the future.
+**12**| `needLists`            |CustomerUser[]                    |List that the user wants to provide the product but is not available.
+**13**| `cartRows`             |CartRow[]                         |Specifies in which basket the product is found.
+**14**| `tags`                 |Tag[]                             |Product tags.
+**15**| `badges`               |Badge[]                           |Badges that are displayed on the product.
+**16**| `review`               |Review                            |Product reviews that include text from user reviews as well as the rates they have given to the product.
+**17**| `customerMetaCategor`  |CustomerMetaCategory              |Information the customer needs to provide to buy a particular product.
+**18**| `discountGroup`        |DiscountGroup                     |Plan of reducing the prices of the products.
+**19**| `priority`             |integer                           |Indicates the priority of the product stored within the system.
+**20**| `productPackage`       |ProductPackage                    |:question:
```

Comment:
If there is a mixture of products creating a new product, we create a product package. So if one of the products in the database has a relation with a ProductPackage, it's understood that the product is a mixture.